src/coreclr/vm/memberload.cpp

```diff
@@ -1132,7 +1133,8 @@ MemberLoader::FindMethod(
             ||
             (pCurDeclMD->MightHaveName(targetNameHash)
             // This is done last since it is the most expensive of the IF statement.
-            && StrCompFunc(pszName, pCurDeclMD->GetName()) == 0)
+            && (pCurDeclMDName = pCurDeclMD->GetName()) != NULL
```

Comment:
This can only return null for corrupted metadata. It may be better to add `MethodDesc::GetNameThrowing` that throws when the metadata API returns error instead of trying to compensate for NULL here.

Is it expected that the metadata is corrupted in the repro that you are looking at?

Reply:
Thank you for the feedback.

> This can only return null for corrupted metadata. It may be better to add MethodDesc::GetNameThrowing that throws when the metadata API returns error instead of trying to compensate for NULL here.
> Is it expected that the metadata is corrupted in the repro that you are looking at?

It occurred once on the production line and the dump is passed to me.
So I think it is just normal execution scenario.